api/lib/infrastructure/repositories/schooling-registration-repository.js

```diff
@@ -55,11 +56,24 @@ module.exports = {
     const trx = await Bookshelf.knex.transaction();
 
     try {
+      const nationalStudentIdsFromFile = schoolingRegistrationDatas.map((schoolingRegistrationData) => schoolingRegistrationData.nationalStudentId);
+      const students = await studentRepository.findReconciledStudentsByNationalStudentId(nationalStudentIdsFromFile);
 
       const schoolingRegistrations = _.map(schoolingRegistrationDatas, (schoolingRegistrationData) => new SchoolingRegistration({ ...schoolingRegistrationData, organizationId }));
       const schoolingRegistrationsFromOrganization = await this.findByOrganizationId({ organizationId });
-      const nationalStudentIdsFromOrganization = _.map(schoolingRegistrationsFromOrganization, 'nationalStudentId');
-      const [ schoolingRegistrationsToUpdate, schoolingRegistrationsToCreate ] = _.partition(schoolingRegistrations, (schoolingRegistration) => _.includes(nationalStudentIdsFromOrganization, schoolingRegistration.nationalStudentId));
+
+      const [ schoolingRegistrationsToUpdate, schoolingRegistrationsToCreate ] = _.partition(schoolingRegistrations, (schoolingRegistration) => {
+        const schoolingRegistrationFromOrganization = schoolingRegistrationsFromOrganization.find((schoolingRegistrationFromOrganization) => schoolingRegistrationFromOrganization.nationalStudentId === schoolingRegistration.nationalStudentId);
+        if (!schoolingRegistrationFromOrganization || (schoolingRegistrationFromOrganization && !schoolingRegistrationFromOrganization.userId)) {
+          if (students) {
+            const student = students.find((student) => student.nationalStudentId === schoolingRegistration.nationalStudentId);
+            if (student) {
+              schoolingRegistration.userId = student.getMostRelevantAccountForReconciliation().userId;
+            }
+          }
+        }
+        return !!schoolingRegistrationFromOrganization;
```

Comment:
C'est un peu :exploding_head: ce double bang
C'est vraiment [le seul moyen ?](https://medium.com/@edplatomail/js-double-bang-or-the-not-operator-part-40e55d089bf0)
> To be honest, I had some difficulty thinking up some good examples for !!.